index.js

```diff
@@ -404,11 +404,6 @@ function compare(buffer1, start1, buffer2, start2) {
       buffer1.readInt32LE(start1 + len1) - buffer2.readInt32LE(start2 + len2)
     )
 
-  //except for strings, sort shorter values first
-  if (type1 !== STRING) {
-    var result = type1 - type2
-    if (result) return result
-  }
```

Comment:
If you look at line 395, then it seems that when the control flow reaches line 408, then we for sure have `type1 === type2` so it doesn't make sense to subtract in line 409 because line 410 will never trigger.

Reply:
right